pkg/datapath/linux/config/config.go

```diff
@@ -549,6 +555,63 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 	return fw.Flush()
 }
 
+// vlanFilterMacros generates VLAN_FILTER macros which
+// are written to node_config.h
+func vlanFilterMacros() (string, error) {
+	devices := make(map[int]bool)
+	for _, device := range option.Config.Devices {
+		ifindex, err := link.GetIfIndex(device)
+		if err != nil {
+			return "", err
+		}
+		devices[int(ifindex)] = true
+	}
+
+	allowedVlans := make(map[int]bool)
+	for _, vlanId := range option.Config.BPFAllowVLAN {
+		allowedVlans[vlanId] = true
```

Comment:
One way to make the "allow all" logic a bit nicer would be to just support "all" as one of the options for the flag. If "all" shows up in this list then rather than iterating the devices and generating a list of specific VLANs to pass through, we can generate `return CTX_ACT_OK` if any VLAN is set.